tests/notnull_tests.cpp

```diff
@@ -181,8 +180,11 @@ SUITE(NotNullTests)
 
     TEST(TestNotNullSharedPtrComparison)
     {
-        auto sp1 = std::make_shared<int>(1);
-        auto sp2 = std::make_shared<const int>(3);
+        int ints[2] = {42, 43};
+        auto emptyDeleter = [](const int*){};
+
+        shared_ptr<int> sp1(&ints[0], emptyDeleter);
+        shared_ptr<const int> sp2(&ints[1], emptyDeleter);
```

Comment:
This change to the `shared_ptr` tests is not necessary for correctness: `shared_ptr` comparisons are defined in terms of `owner_before`, which is defined in terms of `less`, which is total (valid for all pointer values). That said, the change doesn't make it incorrect.